### ISSUE TITLE
fresh crashes

### ISSUE THREAD

**Issue by kambala-decapitator** [xbmc/Official-Kodi-Remote-iOS#604]
archive contains both live and the second "test" build. [Archive.zip](https://github.com/xbmc/Official-Kodi-Remote-iOS/files/8097166/Archive.zip)

also feedback from the user who sent crash from iPad:

iPad iOS 15+ Touching options twice crash the kodi remote app and brings you back to the Home Screen.

What I miss, would like to see and like in the beta app.

* I would love to see the ability to change the order and hide items on the left main menu. (later the user wrote that he found this ability in Settings)
* I would like to see all the ratings I have in my .nfo file for a movie not just IMDb, i.e. rotten tomatoes, omdbapi, trakt, metacritic, IMDb, etc.
* I would like movie information screen to access the .nfo file and show 4K, HD 1080, HD 720, SD
* I miss how the old remote looked. I find it very small now in the beta app.
* I like favorites being added the the left main menu
* I really like that the bottom information button shows the system data i.e. core %, etc. great idea guys!
* I like the access remote button/shortcut on the bottom.
* I really like that installing the beta version did not delete any data!

**Comment by wutschel** [comment]
On the crashes:

- 1.9 and 1.10, `StackScrollViewController addViewInSlider:invokeByController:isStackStartView:` in call `[viewControllersStack removeObjectAtIndex:indexOfViewController];`
-> Fixed (out-of-bounds access caused by wrong index).
- 1.9, `processTrailerFromString:` in call `NSURLComponents componentsWithURL:url resolvingAgainstBaseURL:YES`
-> Fixed (do not call when `url` is `nil`).
- 1.9, Fault hit memory shortage
-> Is this something we can do about?
- 1.10, `checkDiskCache`: in call `mainMenu *menuItem = self.detailItem; NSDictionary *parameters = [Utilities indexKeyedDictionaryFromArray:[menuItem mainParameters][choosedTab]];`
-> Not sure yet. Will be related to the access to `menuItem`.

**Comment by wutschel** [comment]
> * I would love to see the ability to change the order and hide items on the left main menu. (later the user wrote that he found this ability in Settings)

Yes, the items can be already hidden via the app settings. Moving is not possible yet, and I am not sure this adds too much value.

> * I would like to see all the ratings I have in my .nfo file for a movie not just IMDb, i.e. rotten tomatoes, omdbapi, trakt, metacritic, IMDb, etc.

Not sure, if the API supports this.

> * I would like movie information screen to access the .nfo file and show 4K, HD 1080, HD 720, SD

This is possible via the NowPlayingInfo (tapping on the movie cover while playback). As far as I know the same is not possible via the database.

> * I miss how the old remote looked. I find it very small now in the beta app.

That was the kind feedback I was hoping to get from iPad users. If I see correct this was an iPad 7th Gen 10.2-inch. Let me compare the size and make up my mind.

> * I like favorites being added the the left main menu
> * I really like that the bottom information button shows the system data i.e. core %, etc. great idea guys!
> * I like the access remote button/shortcut on the bottom.

Thanks for the positive feedback! :)

> * I really like that installing the beta version did not delete any data!

With next versions this might be the case, as the cached data and the App should stay in sync to not result in inconsistently shown data or even crashes.

**Comment by wutschel** [comment]
@kambala-decapitator, I am still not really clear about the call stack which shows the crash in `checkDiskCache`. This can only be called from 3 methods: `(IBAction)changeTab:(id)sender`, `(void)viewDidLoad` and `(void)handleEnterForeground:(NSNotification*)sender`. I see none them in call stack. Is there something obvious for you which helps narrowing this down?

In case you would relate this to `handleEnterForeground`: I am wondering if `handleEnterForeground` should at all call `[self checkDiskCache]`. `handleEnterForeground` is only triggered by `handleProximityChangeNotification` or when manually enabling the TCP connection in the warning popup which appears, if TCP connection is not enabled in Kodi server. This is not all related to the cache settings for a certain view.

**Comment by kambala-decapitator** [comment]
> * 1.9, Fault hit memory shortage
>   -> Is this something we can do about?

it seems to me that this crash is more about watchdog: probably on app resume main thread was blocked by something, so OS killed it.

> With next versions this might be the case, as the cached data and the App should stay in sync to not result in inconsistently shown data or even crashes.

deleting user data is way worse than deleting caches, cache is ok to be invalidated at some points

> I see none them in call stack

`-[UIApplication _sendWillEnterForegroundCallbacks]`

> So I guess the call of [self checkDiskCache] inside handleEnterForeground is the problem

yes, you're absolutely right

> I am quite sure the call of [self checkDiskCache] as well as the whole implementation of handleEnterForeground is not needed for the DetailViewController

sounds good to me
